File: 프로그래머스/2/42578. 의상/의상.cpp
#include <iostream>
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    unordered_map<string, int> clothesCount;

    // 종류별 옷 개수 저장
    for (auto& item : clothes) {
        string type = item[1];
        clothesCount[type]++;
    }

    int answer = 1;

    // 각 종류마다 (입는 경우 수 + 안 입는 경우 1)씩 곱함
    for (auto& kv : clothesCount) {
        answer *= (kv.second + 1);
    }

    return answer - 1; // 아무것도 안 입는 경우 제외
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*; 

class Solution {
    public int solution(String[][] clothes) {
        
        HashMap<String, Integer> variable = new HashMap<>(); 
        for(String[] c : clothes){
            variable.put(c[1], variable.getOrDefault(c[1], 0) + 1); 
        }
        
        int answer = 1;
        List<Integer> nums = new ArrayList<>(variable.values()); 
        for(int i = 0; i < nums.size(); i++){
            answer *= (nums.get(i) + 1);
        }
        
        // 서로 다른 옷의 조합의 수
        return answer - 1;
    }
}